src/utils.c

```diff
@@ -374,3 +374,37 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// Convert a character vector into a single string, e.g. for printing error messages
+// adapted from https://stackoverflow.com/a/58163237
+// make sure to *free* the returned string after use
+char *joinCharVec (SEXP x, const char *sep)
+{
+    char *joined = NULL;                /* pointer to joined string w/sep */
```

Comment:
Please use 2 whitespace indentation, unless rest of the file you are editing uses different indentation.